File: client/src/lib/session.js
import client from './graphql-client'
import gql from 'graphql-tag'

const graphql = {
    createSession: gql`
        mutation createSession($email: String!, $password: String!) {
            createSession(email: $email, password: $password) {
                token
            }
        }
    `
}

export async function createSession(email, password) {
    return client.call(graphql.createSession, { email, password })
}

File: cli/src/hashtrack/session/session.go
package session

import (
	"context"

	"github.com/Laisky/graphql"
)

type creationMutation = struct {
	CreateSession struct {
		Token graphql.String
	} `graphql:"createSession(email: $email, password: $password)"`
}

type CreationPayload = struct {
	Email    string
	Password string
}

func Create(client *graphql.Client, payload CreationPayload) (string, error) {
	var mutation creationMutation
	variables := map[string]interface{}{
		"email":    graphql.String(payload.Email),
		"password": graphql.String(payload.Password),
	}
	err := client.Mutate(context.Background(), &mutation, variables)

	return string(mutation.CreateSession.Token), err
}
